dvc/tree/gs.py

```diff
@@ -143,6 +144,22 @@ def walk_files(self, path_info, **kwargs):
                 continue
             yield path_info.replace(fname)
 
+    def ls(self, path_info, recursive=False, detail=False):
+        assert recursive
+
+        for blob in self.gs.bucket(path_info.bucket).list_blobs(
+            prefix=path_info.path
+        ):
+            if detail:
+                yield {
+                    "type": "file",
+                    "name": blob.name,
+                    "md5": blob.md5_hash.encode().hex(),
```

Comment:
Unrelated to this PR since we've been using it for some time, but `md5_hash` is not supported for composite objects :slightly_frowning_face: And I don't remember when we did that switch from etag and why :slightly_frowning_face: